shell/common/engine.cc

```diff
@@ -127,18 +131,80 @@ void Engine::OnViewportMetricsChanged(sky::ViewportMetricsPtr metrics) {
     runtime_->SetViewportMetrics(viewport_metrics_);
 }
 
-void Engine::OnLocaleChanged(const mojo::String& language_code,
-                             const mojo::String& country_code) {
-  language_code_ = language_code;
-  country_code_ = country_code;
-  if (runtime_)
-    runtime_->SetLocale(language_code_, country_code_);
+void Engine::DispatchPlatformMessage(
+    ftl::RefPtr<blink::PlatformMessage> message) {
+  if (message->channel() == kLifecycleChannel) {
+    if (HandleLifecyclePlatformMessage(message.get()))
+      return;
+  } else if (message->channel() == kLocalizationChannel) {
+    if (HandleNavigationPlatformMessage(std::move(message)))
+      return;
+  }
+
+  if (runtime_) {
+    runtime_->DispatchPlatformMessage(std::move(message));
+    return;
+  }
+
+  // If there's no runtime_, we need to buffer some navigation messages.
+  if (message->channel() == kNavigationChannel)
+    HandleNavigationPlatformMessage(std::move(message));
```

Comment:
But we drop the locale on the floor if we get the message before the engine is ready.